packages/delegation-process/src/services/delegationConsumerService.ts

```diff
@@ -171,6 +174,73 @@ export function delegationConsumerServiceBuilder(
         )
       );
     },
+
+    async revokeConsumerDelegation(
+      delegationId: DelegationId,
+      { authData, logger, correlationId }: WithLogger<AppContext>
+    ): Promise<void> {
+      const delegatorId = unsafeBrandId<TenantId>(authData.organizationId);
+      logger.info(
+        `Revoking delegation ${delegationId} by consumer ${delegatorId}`
+      );
+
+      const { data: delegation, metadata } = await retrieveDelegationById(
+        readModelService,
+        delegationId
+      );
+
+      assertDelegationKindIs(delegationKind.delegatedConsumer, delegation);
+      assertIsDelegator(delegation, delegatorId);
+      assertIsState(activeDelegationStates, delegation);
+
+      const [delegator, delegate, eservice] = await Promise.all([
+        retrieveTenantById(readModelService, delegation.delegatorId),
+        retrieveTenantById(readModelService, delegation.delegateId),
+        retrieveEserviceById(readModelService, delegation.eserviceId),
```

Comment:
These error are missing on the mapper

Reply:
It's ok that these are mapped to `500` I think, because it's unexpected that a delegation can exist and be revoked with the corresponding tenants/eservice not existing in the readmodel and shall never happen